std/algorithm/iteration.d

```diff
@@ -4257,6 +4267,11 @@ struct Permutations(Range)
 
         next(2);
     }
+
+    @property bool empty() pure nothrow @safe
```

Comment:
Please just use attribute inference. Explicit attributes are nice for documentation but we don't generally document range primitives.